### PR TITLE
[cker] Fix computing MSE Gradient

### PR DESCRIPTION
**Pull request by ragmani**
This commit fixes computing MSE Gradients.
  - SUM_OVER_BATCH_SIZE : computes MSE gradients with sum over batch size as it is before
  - SUM : computes MSE gradients with sum for each batch

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

**Comment by ragmani** [comment]
- For #13736
- Draft #13934

**Review by jyoungyun — APPROVED** [review]
LGTM

FYI, the loss shape varies depending on the reduction type. The `None` reduction type returns the loss to Shape{batch_size} unlike other types. Please keep this in mind this when implementing the user API. :)

File: runtime/onert/backend/train/ops/OperationUtils.cc [hunk 1]
```diff
@@ -97,6 +97,21 @@ void biasGrad(const IPortableTensor *input_backprop, IPortableTensor *bias_grad)
                                            bias_grad_buffer, bias_grad_shape);
 }
 
+nnfw::cker::train::LossReductionType convertLossReductionType(ir::train::LossReductionType type)
+{
+  switch (type)
+  {
+    case ir::train::LossReductionType::SumOverBatchSize:
+      return nnfw::cker::train::LossReductionType::SUM_OVER_BATCH_SIZE;
+      break;
```
Comment (jyoungyun) [inline runtime/onert/backend/train/ops/OperationUtils.cc:106]:
```suggestion
```

File: runtime/onert/backend/train/ops/OperationUtils.cc [hunk 2]
```diff
@@ -97,6 +97,21 @@ void biasGrad(const IPortableTensor *input_backprop, IPortableTensor *bias_grad)
...
+    case ir::train::LossReductionType::Sum:
+      return nnfw::cker::train::LossReductionType::SUM;
+      break;
```
Comment (jyoungyun) [inline runtime/onert/backend/train/ops/OperationUtils.cc:109]:
```suggestion
```

**Comment by ragmani** [comment]
@jyoungyun 

> FYI, the loss shape varies depending on the reduction type. The None reduction type returns the loss to Shape{batch_size} unlike other types. Please keep this in mind this when implementing the user API. :)

Currently, it seems that None reduction type is not supported and loss Shape is always {batch_size}.

**Comment by jyoungyun** [comment]
@ragmani 

> Currently, it seems that None reduction type is not supported and loss Shape is always {batch_size}. 

Yes, the internal implementation has Shape{batch_size}. However, the `get_loss` api returns Shape{1}. It has nothing to do with this, but I just want to remind you one more time. :)

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by ragmani** [comment]
@jyoungyun 
> Yes, the internal implementation has Shape{batch_size}. However, the `get_loss` api returns Shape{1}. It has nothing to do with this, but I just want to remind you one more time. :)

Thanks for your information. Considering your information, we can keep to implement the user API for loss as it is.

**Pull request merged**